src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2293,7 +2293,7 @@ because xbuild doesn't support framework reference assemblies.
 
 <Target Name="_GenerateJavaStubs"
   DependsOnTargets="_SetLatestTargetFrameworkVersion;_PrepareAssemblies;$(_AfterPrepareAssemblies)"
-  Inputs="$(MSBuildAllProjects);@(_ResolvedAssemblies);$(_AndroidManifestAbs);$(_AndroidBuildPropertiesCache);@(_AndroidResourceDest)"
+  Inputs="$(MSBuildAllProjects);@(_ResolvedAssemblies->'%(ReferenceAssembly)');$(_AndroidManifestAbs);$(_AndroidBuildPropertiesCache);@(_AndroidResourceDest)"
```

Comment:
This optimization cannot be performed.  It's entirely valid for an assembly to declare non-`public` types which require Java Callable Wrappers:

```csharp
internal class Example : Java.Lang.Object {
}
```

`_GenerateJavaStubs` needs help, but in a different way.

Reply:
Removed the changes to `_GenerateJavaStubs`...

There may yet be a target that can take advantage of `$(ProduceReferenceAssembly)` and be skipped. I will keep an eye out for one.